packages/gatsby-plugin-feed/src/gatsby-node.js

```diff
@@ -23,27 +23,45 @@ exports.onPostBuild = async ({ graphql }, pluginOptions) => {
    * Run the site settings query to gather context, then
    * then run the corresponding feed for each query.
    */
-  const { query, setup, feeds, ...rest } = {
+  const options = {
     ...defaultOptions,
     ...pluginOptions,
+  } 
+
+  if ('query' in options) {
+    options.query = await runQuery(graphql, options.query)
   }
 
-  const globals = await runQuery(graphql, query)
+  const feeds = options.feeds && Array.isArray(options.feeds) ?
+    options.feeds : [options.output]
 
   for (let f of feeds) {
-    let locals = {}
     if (f.query) {
-      locals = await runQuery(graphql, f.query)
+      f.query = await runQuery(graphql, f.query)
+
+      if (options.query) {
+        f.query = mergeDeep(options.query, f.query)
+        delete options.query
+      }
+    }
+
+    const {
+      setup,
+      ...locals,
+    } = {
+      ...options,
+      ...f,
     }
 
-    const output = path.join(publicPath, f.output)
-    const ctx = { ...globals, ...locals }
-    const feed = new RSS(setup({ ...rest, ...ctx }))
-    const items = f.serialize ? f.serialize(ctx) : serialize(ctx)
+    console.log(locals)
```

Comment:
Good catch. I should probably standardize on the `debug` module for things like this.